server/repository/src/db_diesel/period_schedule.rs

```diff
@@ -0,0 +1,106 @@
+use super::{
+    period_schedule_row::{period_schedule, period_schedule::dsl as period_schedule_dsl},
+    DBType, PeriodScheduleRow, StorageConnection,
+};
+use diesel::prelude::*;
+
+use crate::{
+    diesel_macros::{apply_equal_filter, apply_sort_no_case},
+    repository_error::RepositoryError,
+};
+
+use crate::{EqualFilter, Sort};
+
+pub type PeriodSchedule = PeriodScheduleRow;
+
+#[derive(Clone, PartialEq, Debug)]
+pub struct PeriodScheduleFilter {
+    pub id: Option<EqualFilter<String>>,
+    pub name: Option<EqualFilter<String>>,
+}
+
+#[derive(PartialEq, Debug)]
+pub enum PeriodScheduleSortField {
+    Id,
+    Name,
+}
+
+pub type PeriodScheduleSort = Sort<PeriodScheduleSortField>;
+
+pub struct PeriodScheduleRepository<'a> {
+    connection: &'a StorageConnection,
+}
+
+impl<'a> PeriodScheduleRepository<'a> {
+    pub fn new(connection: &'a StorageConnection) -> Self {
+        PeriodScheduleRepository { connection }
+    }
+
+    pub fn count(&self, filter: Option<PeriodScheduleFilter>) -> Result<i64, RepositoryError> {
+        let query = create_filtered_query(filter);
+        Ok(query.count().get_result(&self.connection.connection)?)
+    }
+
+    pub fn query_by_filter(
+        &self,
+        filter: PeriodScheduleFilter,
+    ) -> Result<Vec<PeriodSchedule>, RepositoryError> {
+        self.query(Some(filter), None)
+    }
+
+    pub fn query(
+        &self,
+        filter: Option<PeriodScheduleFilter>,
+        sort: Option<PeriodScheduleSort>,
+    ) -> Result<Vec<PeriodSchedule>, RepositoryError> {
+        let mut query = create_filtered_query(filter);
+        if let Some(sort) = sort {
+            match sort.key {
+                PeriodScheduleSortField::Id => {
+                    apply_sort_no_case!(query, sort, period_schedule_dsl::id)
+                }
+                PeriodScheduleSortField::Name => {
+                    apply_sort_no_case!(query, sort, period_schedule_dsl::name)
+                }
+            }
+        } else {
+            query = query.order(period_schedule_dsl::name.asc())
```

Comment:
Do we normally put in a default order?
Not sure it's worth worrying about, but it's generally  more efficient for the database to get the database un-ordered unless we actually need it to be ordered in the response.